src/NumericalAlgorithms/RootFinding/RootBracketing.hpp

```diff
@@ -277,12 +286,20 @@ void bracket_possibly_undefined_function_in_interval(
             "bracket_possibly_undefined_function_in_interval: found "
             "case that should not happen under our assumptions.");
       }
-      std::array<double, 4> tmp = bracketing_detail::bracket_by_contracting(
-          {{x3, x1, x2}}, {{y3, y1, y2}}, f);
-      x1 = tmp[0];
-      x2 = tmp[1];
-      y1 = tmp[2];
-      y2 = tmp[3];
+      try {
+        std::array<double, 4> tmp = bracketing_detail::bracket_by_contracting(
```

Comment:
Totally up to you, but if this return a `std::tuple` you could do `std::tie(x1, x2, y1, y2)`. I don't know the code well enough to be able to say whether or not that would be an improvement, so completely up to you!

Reply:
Ok, done.